### PR TITLE
Add bounding box query

### PR DESCRIPTION
**Pull request by A-Stangeland**
When testing the bbox query I found that some of the geometries in the basin data are MultiPolygons. I didn't notice this before because the vast majority of the basin geometries are Polygons. I therefore added Multpolygon as a possible type for the geometry field in the pydantic response model.

### CODE REVIEW

File: deforestation_api/dependencies/queryparams.py [hunk 1]
```diff
@@ -1,34 +1,64 @@
...
+        Query(description="First year in the date range to return data for."),
     ] = 2001,
     end_year: Annotated[
         int,
-        Query(description="Last year to return forest cover loss data for."),
+        Query(description="Last year in the data range to return data for."),
     ] = 2022,
-) -> (int, int):
+) -> tuple[int, int]:
     return start_year, end_year
```
Comment (giltinde) [inline deforestation_api/dependencies/queryparams.py]: Could possibly add a check to verify that `start_year <= end_year`

File: deforestation_api/dependencies/queryparams.py [hunk 2]
```diff
@@ -1,34 +1,64 @@
...
+            description="Maximal longitude of the bounding box to retrieve data for."
+        ),
+    ] = None,
+    max_lat: Annotated[
+        float | None,
+        Query(description="Maximal latitude of the bounding box to retrieve data for."),
+    ] = None,
+) -> tuple[float, float] | tuple[float, float, float, float]:
+    if lon is not None and lat is not None:
+        return lon, lat
```
Comment (giltinde) [inline deforestation_api/dependencies/queryparams.py]: Maybe include in the documentation that if both a point and a bbox are provided that the bbox is ignored, or enforce that only a point or a bbox can be provided (using an error)

File: deforestation_api/dependencies/queryparams.py [hunk 3]
```diff
@@ -1,34 +1,64 @@
...
+    bbox = (min_lon, min_lat, max_lon, max_lat)
+    if all(v is not None for v in bbox):
+        return bbox
```
Comment (giltinde) [inline deforestation_api/dependencies/queryparams.py]: Alternatively you could write `if None not in bbox:`

**Review by giltinde — CHANGES_REQUESTED** [review]

File: deforestation_api/dependencies/queryparams.py (hunk 2 above)
Comment (A-Stangeland) [reply deforestation_api/dependencies/queryparams.py]: I added an error if both are included.

**Review by giltinde — APPROVED** [review]
Nice 🤝

**Review by dymbe — APPROVED** [review]

**Pull request merged**